### PR TITLE
Variable based dark color

### PR DESCRIPTION
**Pull request by JiuqingSong**
Add an experimental feature to allow using variable-based dark mode color.

### CODE REVIEW

**Comment by JiuqingSong** [comment]
#1504

File: packages/roosterjs-editor-dom/lib/utils/setColor.ts [hunk 1]
```diff
@@ -30,32 +34,49 @@ export default function setColor(
     color: string | ModeIndependentColor,
     isBackgroundColor: boolean,
     isDarkMode?: boolean,
-    shouldAdaptTheFontColor?: boolean
+    shouldAdaptTheFontColor?: boolean,
+    darkColorHandler?: DarkColorHandler | null
```
Comment (BryanValverdeU) [inline packages/roosterjs-editor-dom/lib/utils/setColor.ts:38]: nit: update the params in the comment.

File: packages/roosterjs-editor-dom/lib/utils/applyFormat.ts [hunk 2]
```diff
@@ -9,7 +9,8 @@ import { DefaultFormat } from 'roosterjs-editor-types';
 export default function applyFormat(
     element: HTMLElement,
     format: DefaultFormat,
-    isDarkMode?: boolean
+    isDarkMode?: boolean,
+    darkColorHandler?: DarkColorHandler | null
```
Comment (BryanValverdeU) [inline packages/roosterjs-editor-dom/lib/utils/applyFormat.ts:13]: nit: update the params in the comment.

File: packages/roosterjs-editor-dom/lib/utils/setColor.ts [hunk 3]
```diff
@@ -64,16 +85,12 @@ export default function setColor(
...
-        (isDarkMode &&
-            (element.dataset[DarkModeDatasetNames.OriginalStyleBackgroundColor] ||
-                element.dataset[DarkModeDatasetNames.OriginalAttributeBackgroundColor])) ||
-        backgroundColor;
+function adaptFontColorToBackgroundColor(
+    element: HTMLElement,
+    lightModeBackgroundColor: string,
+    isDarkMode?: boolean,
+    darkColorHandler?: DarkColorHandler | null
+) {
```
Comment (BryanValverdeU) [inline packages/roosterjs-editor-dom/lib/utils/setColor.ts:93]: nit: update the params in the comment.

**Review by BryanValverdeU — APPROVED** [review]

**Pull request merged**